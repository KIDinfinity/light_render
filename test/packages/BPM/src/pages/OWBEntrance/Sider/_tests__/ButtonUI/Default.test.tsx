import React from 'react';
import renderer from 'react-test-renderer';
import Default from '../../ButtonUI/Default';

jest.mock('../../ButtonUI/Icon', () => require('mocks/OWB/Sider/ButtonUI/Icon'));

describe('test icon', () => {
  test('default', () => {
    const action = () => {};
    const tree = renderer
      .create(
        <Default
          title="default button"
          styles={{
            testIcon: 'testIcon',
            box: 'test-box',
            className: 'testClassName',
          }}
          status="default"
          icon="testIcon"
          buttonCode="testButtonCode"
          action={action}
          className="className"
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
