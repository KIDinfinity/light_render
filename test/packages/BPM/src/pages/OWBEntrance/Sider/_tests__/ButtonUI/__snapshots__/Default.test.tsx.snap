// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test icon default 1`] = `
<button
  className="test-box testIcon testClassName undefined"
  data-role="bpmButton-testButtonCode"
  data-trigger="testButtonCode"
  onClick={[Function]}
  type="button"
>
  <span>
     this is icon
  </span>
  <span>
    default button
  </span>
</button>
`;
